pkg/metrics/metrics.go

```diff
@@ -167,89 +187,36 @@ func BlockMined(block *proto.Block, height proto.Height) {
 		return
 	}
 	t := newTags().withBlock().withEvent(eventMined).withID(block.ID).withParentID(block.Parent).withBroadcast()
-	f := newFields().withHeight(height).withTransactionsCount(block.TransactionCount).withBaseTarget(block.BaseTarget)
+	f := newFields().withHeight(height).withTransactionsCount(block.TransactionCount).withBaseTarget(block.BaseTarget).
+		withID(block.ID)
 	reportBlock(t, f)
 }
 
-func FSMKeyBlockReceived(fsm string, block *proto.Block, source string) {
-	if rep == nil {
-		return
-	}
-	t := emptyTags().node().fsm(fsm).block().received()
-	f := emptyFields().blockID(block.BlockID()).referenceID(block.Parent).source(source).blockTS(block.Timestamp).genPK(block.GeneratorPublicKey)
-	reportFSM(t, f)
-}
-
-func FSMKeyBlockGenerated(fsm string, block *proto.Block) {
-	if rep == nil {
-		return
-	}
-	t := emptyTags().node().fsm(fsm).block().generated()
-	f := emptyFields().blockID(block.BlockID()).referenceID(block.Parent)
-	reportFSM(t, f)
-}
-
-func FSMKeyBlockApplied(fsm string, block *proto.Block) {
-	if rep == nil {
-		return
-	}
-	t := emptyTags().node().fsm(fsm).block().applied()
-	f := emptyFields().blockID(block.BlockID()).referenceID(block.Parent)
-	reportFSM(t, f)
-}
-
-func FSMKeyBlockDeclined(fsm string, block *proto.Block, err error) {
-	if rep == nil {
-		return
-	}
-	t := emptyTags().node().fsm(fsm).block().declined()
-	f := emptyFields().blockID(block.BlockID()).referenceID(block.Parent).error(err)
-	reportFSM(t, f)
-}
-
-func FSMMicroBlockReceived(fsm string, microblock *proto.MicroBlock, source string) {
-	if rep == nil {
-		return
-	}
-	t := emptyTags().node().fsm(fsm).microblock().received()
-	f := emptyFields().blockID(microblock.TotalBlockID).referenceID(microblock.Reference).source(source)
-	reportFSM(t, f)
-}
-
-func FSMMicroBlockGenerated(fsm string, microblock *proto.MicroBlock) {
-	if rep == nil {
-		return
-	}
-	t := emptyTags().node().fsm(fsm).microblock().generated()
-	f := emptyFields().blockID(microblock.TotalBlockID).referenceID(microblock.Reference).signature(microblock.TotalResBlockSigField)
-	reportFSM(t, f)
-}
-
-func FSMMicroBlockDeclined(fsm string, microblock *proto.MicroBlock, err error) {
+func MicroBlockMined(mb *proto.MicroBlock) {
```

Comment:
Here we should report total transactions count of the block, not micro-block. Looks like we have to add separate parameter here.